presto-tests/src/test/java/io/prestosql/execution/EventsAwaitingQueries.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import io.prestosql.Session;
+import io.prestosql.testing.MaterializedResult;
+import io.prestosql.testing.QueryRunner;
+import org.intellij.lang.annotations.Language;
+
+import java.util.Optional;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.fail;
+
+class EventsAwaitingQueries
```

Comment:
I don't like this class, I would inline it. It looks much simpler to use `AbstractTestFramework` or `QueryAssertions` (preferable) to verify queries, where events could be handled manually. Something like we do `io.prestosql.sql.query.TestRowFilter`.

Reply:
It's a wrapper around custom written code to execute a query and block until all expected events are gathered. It's extracted from existing tests so it's not really a case for QueryAssertions etc. I could not extract it and add this test to the other tests, but then the setup for it would grow even bigger